zaza/openstack/charm_tests/ceph/rbd_mirror/tests.py

```diff
@@ -166,6 +184,18 @@ def setUpClass(cls):
         # get ready for multi-model Zaza
         cls.site_a_model = cls.site_b_model = zaza.model.get_juju_model()
         cls.site_b_app_suffix = '-b'
+        # Test if we have the needed Openstack applications.
+        try:
+            zaza.model.get_application(cls.cinder_ceph_app_name)
+            cls.with_cinder = True
```

Comment:
This is caching whether the `cinder-ceph` application is deployed in the model at the beginning of the tests.  As caching is one of the hard problems, perhaps what's really needed is a predicate function that can be called at each test.

e.g. if some future test decided to test what happens when cinder-ceph is removed from the model (I don't why!) then this caching would break.  I think we should all develop more a mindset around avoiding caching where possible unless it really is a performance or memory issue.

Reply:
I understand where you're coming from. In the context of this PR, this caching is only looked at the beginning of a couple of tests to skip them in an application is not present. It's pretty harmless in my opinion, but I can accommodate further changes. For example, I can document how it's supposed to be used or replace the caching for a call that looks for the application on the model on each run. What do you think?

Reply:
Yes, I think documenting it in the code would provide enough sign-posting if you want to keep the cached value; however, if you can avoid caching altogether then it would make the code more robust from a maintenance perspective.  If there are performance issues, or other undesirable side-effects, then caching is acceptable.

Reply:
Done, there's no longer caching.